Calculator/Calculator/Model/Extensions/String+.swift

```diff
@@ -0,0 +1,14 @@
+//
+//  String+.swift
+//  Calculator
+//
+//  Created by Serena on 2023/05/31.
+//
+
+extension String: CalculateItem {
+    func split(with target: Character) -> [String] {
+        let targetArray = self.split(separator: target).map { String($0) }
```

Comment:
여기서 `map`에 전달되는 `transform` 매개변수의 경우 `(Self.Element) throws -> T` 함수 타입인데요! 그 말은 해당 조건(전달인자를 받아 적절한 타입으로 반환할 것)을 만족하는 함수나 메서드를 전달할 수 있다는 것입니다. 그래서 다음과 같이 작성해볼 수도 있습니다.

```suggestion
        let targetArray = self.split(separator: target).map(String.init)
```

고차함수를 사용할 때 단순히 특정 함수나 메서드를 호출하는 것이라면 이러한 방법을 고려해볼 수도 있습니다.

라는 코멘트를 달고 `ExpressionParser`를 보니 `numberParts`라는 변수를 선언할 때는 그렇게 작성하셨더라고요.

`init`을 전달할 것인지 안 할 것인지 컨벤션의 통일이 필요합니다.

Reply:
string+를 refactor하는 것을 놓쳤습니다🥲 가독성을 위해 (Type.init)으로 컨벤션 통일하겠습니다!